lib/src/back_end/solution.dart

```diff
@@ -112,8 +154,19 @@ class Solution implements Comparable<Solution> {
   /// code.
   @override
   int compareTo(Solution other) {
-    var scoreComparison = score.compareTo(other.score);
-    if (scoreComparison != 0) return scoreComparison;
+    // TODO(perf): This check is necessary to not make some of the tests not
+    // timeout, but I'm not sure why. Investigate what this does and if it's
+    // still necessary once other optimizations are in place.
+    // An invalid solution is worse than all others.
+    if (!isValid) return 1;
```

Comment:
Should we try to make this symmetric even when comparing two invalid solutions?

Reply:
Actually, it looks like this check is no longer needed at all now that I've optimized it. Removed the validity checks completely.